include/wx/clntdata.h

```diff
@@ -158,5 +159,38 @@ class WXDLLIMPEXP_BASE wxClientDataContainer
 
 };
 
+// This class is a replacement for wxClientDataContainer, and unlike
+// wxClientDataContainer the wxSharedClientDataContainer client data is
+// possible to copy (as a shared ptr) when instances of it are cloned.
+// Like wxClientDataContainer, wxSharedClientDataContainer is a mixin
+// that provides storage and management of "client data.". The client data
+// is reference counted and managed by the container.
+//
+// NOTE:  If your class has a clone function and needs to store client data,
+//        use wxSharedClientDataContainer and not wxClientDataContainer!
+
+class WXDLLIMPEXP_BASE wxSharedClientDataContainer
+{
+public:
```

Comment:
I'd add a comment here:
```suggestion
public:
    // Provide the same functions as in wxClientDataContainer, so that objects
    // using it and this class could be used in exactly the same way.
```